### DISCUSSION TITLE
ScriptReference/Mathf.SmoothDamp

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#613] [Notes]
# ScriptReference/Mathf.SmoothDamp

https://docs.unity3d.com/ScriptReference/Mathf.SmoothDamp.html

**Comment by naelstrof** [comment]
Mathf.SmoothDamp will generate a NaN if Time.deltaTime is zero. Even if it's not supplied. 
Best to use it within a guard if your game includes a pause feature:
```csharp
void Update() {
    if (Time.deltaTime != 0f) {
        newPosition = Mathf.SmoothDamp(newPosition, oldPosition, ref velocity);
    }
}
```
